tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,240 @@
+			"istios.operator.istio.io",
+			"peerauthentications.security.istio.io",
+			"proxyconfigs.networking.istio.io",
+			"requestauthentications.security.istio.io",
+			"serviceentries.networking.istio.io",
+			"sidecars.networking.istio.io",
+			"telemetries.telemetry.istio.io",
+			"virtualservices.networking.istio.io",
+			"wasmplugins.extensions.istio.io",
+			"workloadentries.networking.istio.io",
+			"workloadgroups.networking.istio.io",
+		}
+	)
+
+	Describe("installation", func() {
+		// TODO: we  need to support testing both types of deployment for the operator, helm and olm via subscription.
+		// Discuss with the team if we should add a flag to the test to enable the olm deployment and don't do that deployment in different step
+		When("installed via helm install", func() {
+			BeforeAll(func() {
+				if skipDeploy == "true" {
+					Skip("Skipping the deployment of the operator and the tests")
+				}
+
+				Expect(kubectl.CreateNamespace(namespace)).To(Succeed(), "Namespace failed to be created; unexpected error")
+
+				extraArg := ""
+				if ocp == "true" {
+					extraArg = "--set=platform=openshift"
+				}
+
+				installArgs := fmt.Sprintf("--namespace %s --set=image=%s %s", namespace, image, extraArg)
+				Eventually(helm.Install).
+					WithArguments("sail-operator", filepath.Join(baseDir, "chart"), installArgs).
+					Should(Succeed(), "Operator failed to be deployed; unexpected error")
+			})
+
+			It("starts successfully", func() {
+				Eventually(kubectl.GetJSONCondition).
+					WithArguments(namespace, "deployment", deploymentName).
+					Should(ContainElement(resourceAvailable))
+				Success("Operator deployment is Available")
+
+				Expect(kubectl.GetPodPhase(namespace, "control-plane=istio-operator")).Should(Equal("Running"), "Operator failed to start; unexpected pod Phase")
+				Success("Istio-operator pod is Running")
+			})
+
+			It("deploys all the CRDs", func() {
+				Eventually(kubectl.GetCRDs).
+					Should(ContainElements(istioResources), "Istio CRDs are not present; expected list to contain all elements")
+				Success("Istio CRDs are present")
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio resource", func() {
+		for _, version := range istioVersions {
+			// Note: This var version is needed to avoid the closure of the loop
+			version := version
+			var err error
+
+			Context(fmt.Sprintf("version %s", version), func() {
+				BeforeAll(func() {
+					Expect(kubectl.CreateNamespace(controlPlaneNamespace)).To(Succeed(), "Namespace failed to be created; unexpected error")
+					istioCR = v1alpha1.Istio{
+						TypeMeta: metav1.TypeMeta{
+							APIVersion: "operator.istio.io/v1alpha1",
+							Kind:       "Istio",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name: istioName,
+						},
+						Spec: v1alpha1.IstioSpec{
+							Namespace: controlPlaneNamespace,
+							Version:   version,
+						},
+					}
+
+					istioCRYAML, err = yaml.Marshal(istioCR)
```

Comment:
I think I'd prefer to see this as a plain YAML string, since that's what users will be deploying. It would also make the test simpler and would fail if we end up renaming a field, which is exactly what we want (i.e. the test should fail if we make a change to the Istio schema).

Reply:
Done